org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/XMLSchemaErrorCode.java

```diff
@@ -185,6 +189,20 @@ public static Range toLSPRange(XMLLocator location, XMLSchemaErrorCode code, Obj
 				} else {
 					SchemaLocation schemaLocation = document.getSchemaLocation();
 					if (schemaLocation != null) {
+						String invalidSchemaPath = arguments[0] instanceof String ? (String) arguments[0] : null;
+						
+						if (invalidSchemaPath != null) {
+							for (SchemaLocationHint locHintRange : schemaLocation.getSchemaLocationHints()) {
+								try {
+									String expandedHint = XMLEntityManager.expandSystemId(locHintRange.getHint(), document.getDocumentURI(), false);

```

Comment:
create a private static methd getResolvedLocation (it should help to fix https://github.com/eclipse/lemminx/issues/801)